brainiak/io.py

```diff
@@ -0,0 +1,105 @@
+#  Copyright 2017 Intel Corporation
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""I/O functionality."""
+
+from pathlib import Path
+from typing import Iterable, List, Union
+
+import nibabel as nib
+import numpy as np
+
+from nibabel.nifti1 import Nifti1Pair
+from nibabel.spatialimages import SpatialImage
+
+from .image import SingleConditionSpec
+
+
+def load_images_from_dir(in_dir: Union[str, Path], suffix: str = "nii.gz",
+                         ) -> Iterable[SpatialImage]:
+    """Load images from directory.
+
+    A new generator must be created by calling the function for every new
+    iteration through the images.
+
+    Parameters
+    ----------
+    in_dir:
+        Path to directory.
+    suffix:
+        Only load images with names that end like this.
+
+    Yields
```

Comment:
Done. Regarding, sklearn, I think it is not relevant, because the caller must further process the images anyway to construct `np.ndarray`s to pass to sklearn objects. If you meant the docstring standard, "Yields" is conforming to the Sphinx NumPy-style standard.